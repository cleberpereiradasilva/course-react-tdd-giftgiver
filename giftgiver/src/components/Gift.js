import React, { Component } from 'react'
import { 
    Form, 
    FormGroup, 
    FormControl, 
    ControlLabel,
    Col,
    Row,
    Button 
} from 'react-bootstrap'

class Gift extends Component{

    constructor(props){
        super(props)
        this.state = { person: '', present: ''}
    } 

    render(){
        return (
            <Row className='gift'>
                <Form className='gift__form'>
                    <div className='row'>
                        <Col md={5}>
                            <FormGroup>
                                <ControlLabel>Person</ControlLabel>
                                <FormControl className='input-person'                         
                                    onChange={ e => this.setState({ person: e.target.value }) } />                                       
                            </FormGroup>
                        </Col>
                        <Col md={5}>
                        <FormGroup>
                            <ControlLabel>Present</ControlLabel>
                            <FormControl className='input-gift'                         
                                onChange={ e => this.setState({ present: e.target.value }) } />
                        </FormGroup>
                        </Col>
                        <Col md={2} className='gift__form__row__button'>
                            <Button type='button' className='btn btn-danger button-remove' 
                                onClick={() => {
                                    this.props.handleRemove(this.props.gift)
                                }}>
                                X
                            </Button>
                        </Col>
                    </div>
                </Form>
                
            </Row>
        )
    }
}

export default Gift;