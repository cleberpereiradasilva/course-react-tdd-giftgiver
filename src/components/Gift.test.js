import React from 'react'
import { shallow } from 'enzyme'
import Gift from './Gift'


describe("Gift", () => {
    const mockRemove = jest.fn()
    const id = 1
    const props = { gift: id, handleRemove: mockRemove}
    const gift = shallow(<Gift {...props} />)
    it('renders correctly', () => {
        expect(gift).toMatchSnapshot()
    })
    it('initialize `person`', () =>{
        expect(gift.state()).toEqual({person: '', present: ''})
    })
    describe("when typing into person input", () => {
        const person = 'Uncle'
        beforeEach(() => {
            gift.find('.input-person').simulate('change', {
                target : { value: person} 
            })
        })
        it('update `person` in state', () =>{
            expect(gift.state().person).toEqual(person)
        })
    })

    describe("when typing into gift input", () => {
        const present = 'Golf Club'
        beforeEach(() => {
            gift.find('.input-gift').simulate('change', {
                target : { value: present} 
            })
        })
        it('update `present` in state', () =>{
            expect(gift.state().present).toEqual(present)
        })
    })

    describe('when user click in remove a `Remove Gift` button', () =>{
        beforeEach(() => {
            gift.find('.button-remove').simulate('click')
        })
        it('call remove `Gift`', () =>{
            expect(mockRemove).toHaveBeenCalledWith(id)            
        })
    })
})