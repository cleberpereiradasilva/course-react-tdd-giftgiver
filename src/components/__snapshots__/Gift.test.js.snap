// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Row
    bsClass="row"
    className="gift"
    componentClass="div"
>
    <Form
        bsClass="form"
        className="gift__form"
        componentClass="form"
        horizontal={false}
        inline={false}
    >
        <div
            className="row"
        >
            <Col
                bsClass="col"
                componentClass="div"
                md={5}
            >
                <FormGroup
                    bsClass="form-group"
                >
                    <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
                    >
                        Person
                    </ControlLabel>
                    <FormControl
                        bsClass="form-control"
                        className="input-person"
                        componentClass="input"
                        onChange={[Function]}
                    />
                </FormGroup>
            </Col>
            <Col
                bsClass="col"
                componentClass="div"
                md={5}
            >
                <FormGroup
                    bsClass="form-group"
                >
                    <ControlLabel
                        bsClass="control-label"
                        srOnly={false}
                    >
                        Present
                    </ControlLabel>
                    <FormControl
                        bsClass="form-control"
                        className="input-gift"
                        componentClass="input"
                        onChange={[Function]}
                    />
                </FormGroup>
            </Col>
            <Col
                bsClass="col"
                className="gift__form__row__button"
                componentClass="div"
                md={2}
            >
                <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="btn btn-danger button-remove"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                >
                    X
                </Button>
            </Col>
        </div>
    </Form>
</Row>,
  "nodes": Array [
    <Row
      bsClass="row"
      className="gift"
      componentClass="div"
>
      <Form
            bsClass="form"
            className="gift__form"
            componentClass="form"
            horizontal={false}
            inline={false}
      >
            <div
                  className="row"
            >
                  <Col
                        bsClass="col"
                        componentClass="div"
                        md={5}
                  >
                        <FormGroup
                              bsClass="form-group"
                        >
                              <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                              >
                                    Person
                              </ControlLabel>
                              <FormControl
                                    bsClass="form-control"
                                    className="input-person"
                                    componentClass="input"
                                    onChange={[Function]}
                              />
                        </FormGroup>
                  </Col>
                  <Col
                        bsClass="col"
                        componentClass="div"
                        md={5}
                  >
                        <FormGroup
                              bsClass="form-group"
                        >
                              <ControlLabel
                                    bsClass="control-label"
                                    srOnly={false}
                              >
                                    Present
                              </ControlLabel>
                              <FormControl
                                    bsClass="form-control"
                                    className="input-gift"
                                    componentClass="input"
                                    onChange={[Function]}
                              />
                        </FormGroup>
                  </Col>
                  <Col
                        bsClass="col"
                        className="gift__form__row__button"
                        componentClass="div"
                        md={2}
                  >
                        <Button
                              active={false}
                              block={false}
                              bsClass="btn"
                              bsStyle="default"
                              className="btn btn-danger button-remove"
                              disabled={false}
                              onClick={[Function]}
                              type="button"
                        >
                              X
                        </Button>
                  </Col>
            </div>
      </Form>
</Row>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Gift
        gift={1}
        handleRemove={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Gift {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "gift": 1,
          "handleRemove": [Function],
        },
        "refs": Object {},
        "state": Object {
          "person": "",
          "present": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Row
          bsClass="row"
          className="gift"
          componentClass="div"
>
          <Form
                    bsClass="form"
                    className="gift__form"
                    componentClass="form"
                    horizontal={false}
                    inline={false}
          >
                    <div
                              className="row"
                    >
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={5}
                              >
                                        <FormGroup
                                                  bsClass="form-group"
                                        >
                                                  <ControlLabel
                                                            bsClass="control-label"
                                                            srOnly={false}
                                                  >
                                                            Person
                                                  </ControlLabel>
                                                  <FormControl
                                                            bsClass="form-control"
                                                            className="input-person"
                                                            componentClass="input"
                                                            onChange={[Function]}
                                                  />
                                        </FormGroup>
                              </Col>
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={5}
                              >
                                        <FormGroup
                                                  bsClass="form-group"
                                        >
                                                  <ControlLabel
                                                            bsClass="control-label"
                                                            srOnly={false}
                                                  >
                                                            Present
                                                  </ControlLabel>
                                                  <FormControl
                                                            bsClass="form-control"
                                                            className="input-gift"
                                                            componentClass="input"
                                                            onChange={[Function]}
                                                  />
                                        </FormGroup>
                              </Col>
                              <Col
                                        bsClass="col"
                                        className="gift__form__row__button"
                                        componentClass="div"
                                        md={2}
                              >
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="default"
                                                  className="btn btn-danger button-remove"
                                                  disabled={false}
                                                  onClick={[Function]}
                                                  type="button"
                                        >
                                                  X
                                        </Button>
                              </Col>
                    </div>
          </Form>
</Row>,
        "_debugID": 2,
        "_renderedOutput": <Row
          bsClass="row"
          className="gift"
          componentClass="div"
>
          <Form
                    bsClass="form"
                    className="gift__form"
                    componentClass="form"
                    horizontal={false}
                    inline={false}
          >
                    <div
                              className="row"
                    >
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={5}
                              >
                                        <FormGroup
                                                  bsClass="form-group"
                                        >
                                                  <ControlLabel
                                                            bsClass="control-label"
                                                            srOnly={false}
                                                  >
                                                            Person
                                                  </ControlLabel>
                                                  <FormControl
                                                            bsClass="form-control"
                                                            className="input-person"
                                                            componentClass="input"
                                                            onChange={[Function]}
                                                  />
                                        </FormGroup>
                              </Col>
                              <Col
                                        bsClass="col"
                                        componentClass="div"
                                        md={5}
                              >
                                        <FormGroup
                                                  bsClass="form-group"
                                        >
                                                  <ControlLabel
                                                            bsClass="control-label"
                                                            srOnly={false}
                                                  >
                                                            Present
                                                  </ControlLabel>
                                                  <FormControl
                                                            bsClass="form-control"
                                                            className="input-gift"
                                                            componentClass="input"
                                                            onChange={[Function]}
                                                  />
                                        </FormGroup>
                              </Col>
                              <Col
                                        bsClass="col"
                                        className="gift__form__row__button"
                                        componentClass="div"
                                        md={2}
                              >
                                        <Button
                                                  active={false}
                                                  block={false}
                                                  bsClass="btn"
                                                  bsStyle="default"
                                                  className="btn btn-danger button-remove"
                                                  disabled={false}
                                                  onClick={[Function]}
                                                  type="button"
                                        >
                                                  X
                                        </Button>
                              </Col>
                    </div>
          </Form>
</Row>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Gift
    gift={1}
    handleRemove={[Function]}
/>,
}
`;
